actionpack/lib/action_controller/metal/renderers.rb

```diff
@@ -25,9 +42,16 @@ module Renderers
     # Default values are <tt>:json</tt>, <tt>:js</tt>, <tt>:xml</tt>.
     RENDERERS = Set.new
```

Comment:
Was thinking an interesting style would be to define renderers per mime_type or format e.g. it would be something like

Given:

```ruby
content_type = 'application/json'
mime_type = Mime::Type.lookup(content_type)
format = :json # we can't rely on a format passed to render nor that it corresponds to a mime type.
default_renderer = :json
```

then

```ruby
RENDERERS = {
  json: {  # 'format', e.g. `render json: model`
    content_type: Mime[:json], # each format has an associated mime type
    adapters: { # I called this a serializer in the original comment, but am thinking adapter might be better?
      default: "_render_with_renderer_json(json, options)" # this is basically the default json renderer.  Maybe it should be a callable, but this is the idea
    }
  }
}
```

Then, you could `ActionController::Renderers.add :json, custom_adapter: { |json, options| whatever }` and `render json: model, adapter: :custom_adapter` or maybe even `use_adapter json: :custom_adapter`.

To render JSON API it might make sense to add a `json_api` renderer and adapter e.g.

```ruby
RENDERERS = {
  json: {  } # etc, see above
  json_api: {  # 'format', e.g. `render json_api: model`
    content_type: Mime[:json_api], # each format has an associated mime type
    adapters: {
      default: "ActiveModel::SerializableResource.new(json, options)" # this is basically the ActiveModelSerializers serializer
    }
  }
}
```

But I'm not sure I'd want to change my renders from the `render json: model`, so maybe consider that the `json` format, and it uses the `json` renderer (calls `as_json` and sets `content_type`) and `adapter: :json_api` (representation of 'model' that be as_json, if not a string)

```ruby
RENDERERS = {
  json: {  # 'format', e.g. `render json: model`
    adapters: {
      default: {
        content_type: Mime[:json], # each serializer has an associated mime type
        adapter: "_render_with_renderer_json(json, options)" # this is basically the default json renderer.  Maybe it should be a callable, but this is the idea
      },
      json_api:  {
        content_type: Mime[:json_api], # each serializer has an associated mime type
        adapter: "ActiveModel::SerializableResource.new(json, options)" # this is basically the ActiveModelSerializers serializer
      },
    }
  }
}
```

then we could

```ruby
render json: model
# OR
render json: model, adapter: :default
# OR
render json: model, adapter: :json_api
# OR
use_renderer json: :json_api
def show
  render json: model #=== render json: model, adapter: :json_api
end
## AND
use_renderer :json
def show
  render json: model #=== render json: model, adapter: :default
end
```